lib/HTML/Restrict.pm

```diff
@@ -573,6 +597,29 @@ feature is off by default.
     $html = $hr->process( $html );
     # $html is now: "<!-- comments! -->foo"
 
+=item * max_parser_loops => [Integer]
+
+Defaults to 25.  Should never be less than 2.
+
+As of v2.4.0, calling C<process()> will force the parser to clean the text
+multiple times, stopping only once the text is no longer changed or once
+C<max_parser_loops> has been reached.
+
+The reason for this is that L<HTML::Parser> could take malformed HTML and turn
+it into well formed HTML.  This can defeat our processing logic and allow
+malicious input to be returned.  In order to mitigate this, we will clean all
+input at least two times.  If the second attempt at cleaning does not match
+the previous attempt, we will make a third attempt and so on.  This helps to
+ensure that we get the expected output.
+
+If we are unable to get unchanged values after reaching C<max_parser_loops>, an
+exception will be thrown.  Returning partially cleaned text would be wrong, as
+would be returning C<undef> or an empty string.  Throwing an exception forces
+the user to choose the appropriate way of dealing with this.
+
+If you choose to set this value, please note that it can be no less than 2,
+or the parser will never be able to make a comparison with a previous value.
```

Comment:
I wonder if it would be better to enforce this with a type check. Right now, setting it to a number below 2 effectively disables the check and people might come to depend on that.

Reply:
Good idea. I've added a type check.